stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/WriteTextRecord.java

```diff
@@ -56,24 +67,68 @@
     private static final int CONTENT_INDEX = 1;
 
     /**
-     * Writes records for a given file.
+     * Will be the I/O event handler.
+     */
+    private EventManager eventManager = EventManager.getInstance();
+
+    private static final Logger log = LoggerFactory.getLogger(WriteTextRecord.class);
+
+/*
+    private static EventResult writeRecordResponse(EventResult<Integer, EventContext> result) {
+        BStruct errorStruct;
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        return result;
+    }
+*/
+
+    /**
+     * Asynchronously writes records to the channel.
      *
+     * @param recordChannel channel the records should be written.
+     * @param records       the record content.
+     * @param context       event context.
+     * @throws ExecutionException   during interrupt error.
+     * @throws InterruptedException errors which occurs while execution.
+     */
+    private void writeTextRecord(DelimitedRecordChannel recordChannel, BStringArray records, EventContext context)
+            throws ExecutionException, InterruptedException {
+        DelimitedRecordWriteEvent recordWriteEvent = new DelimitedRecordWriteEvent(recordChannel, records, context);
+        CompletableFuture<EventResult> future = eventManager.publish(recordWriteEvent);
+        //future.thenApply(WriteTextRecord::writeRecordResponse);
+        EventResult eventResult = future.get();
+        Throwable error = ((EventContext) eventResult.getContext()).getError();
+        if (error != null) {
+            throw new ExecutionException(error);
+        }
+    }
+
+    /**
+     * Writes records for a given file.
+     * <p>
      * {@inheritDoc}
      */
     @Override
     public BValue[] execute(Context context) {
         BStruct channel;
         BStringArray content;
+        BStruct errorStruct = null;
```

Comment:
Shall we move the variables to the try block here as well? Check all the places

Reply:
We can't move this to the try block since, this is used to initialize the error value